### PR TITLE
[IMP] mail: activity view: allow to filter columns

### PR DESCRIPTION
**Pull request by fdardenne**
Before, in activity view, it was not possible for users to hide never used columns.

Now it is possible for users to hide and show columns in the activity view.

task-3039919

image

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
@qdp-odoo

File: addons/mail/static/src/views/activity/activity_renderer.xml [hunk 1]
```diff
@@ -43,10 +66,12 @@
     <tr class="o_data_row h-100" t-attf-class="{{ activeFilter.resIds.includes(resId) ? 'o_activity_filter_' + activeFilter.progressValue.active : '' }}">
         <t t-set="record" t-value="getRecord(resId)"/>
         <ActivityRecord archInfo="props.archInfo" record="record" openRecord="props.openRecord" />
-        <t t-foreach="props.activityTypes" t-as="type" t-key="type[0]">
+        <t t-foreach="activeColumns" t-as="type" t-key="type[0]">
             <t t-call="mail.ActivityViewCell"/>
         </t>
+        <td></td>
```
Comment (Iucapad) [inline addons/mail/static/src/views/activity/activity_renderer.xml:72]: You could just use `<td/>`

File: addons/mail/static/src/views/activity/activity_renderer.xml [hunk 2]
```diff
@@ -43,10 +66,12 @@
...
     </tr>
+
```
Comment (Iucapad) [inline addons/mail/static/src/views/activity/activity_renderer.xml:74]: remove this empty line, please

**Review by Iucapad — APPROVED** [review]
Just two small comments. Nice job :)

**Comment by Iucapad** [comment]
@robodoo r+

**Comment by fdardenne** [comment]
@robodoo rebase-ff

**Comment by Iucapad** [comment]
@robodoo rebase-ff

**Pull request closed without merging**